### PR TITLE
Clearly identify errors killing pods in events and logs

### PR DESCRIPTION
**Pull request by pmorie**
Whilst debugging a problem with pods stuck in terminating it came to light that the error messages may be ambiguous when killing a pod fails.

@kubernetes/sig-node 
@kubernetes/rh-cluster-infra

---

This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/24743)

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1749,6 +1749,7 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, podStatus *kubecont
 	// Kill pods we can't run.
 	if err := canRunPod(pod); err != nil || pod.DeletionTimestamp != nil {
 		if err := kl.killPod(pod, nil, podStatus); err != nil {
+			err = fmt.Errorf("error killing pod: %v", err)
```
Comment (derekwaynecarr) [inline pkg/kubelet/kubelet.go]: that doesn't change the outer err that we log because of shadowing....

Comment (derekwaynecarr) [inline pkg/kubelet/kubelet.go]: That doesnt change the error that a pod worker uses when it generates the event, I thought that was your goal...

Comment (derekwaynecarr) [inline pkg/kubelet/kubelet.go]: 1750 declares err, 151 redeclares err, you modify that err, but 1755 returns err from 1750!

Comment (pmorie) [inline pkg/kubelet/kubelet.go]: i think you're right, will change

**Comment by pmorie** [comment]
@derekwaynecarr @yujuhong fixed the shadowing problem derek pointed out

**Comment by yujuhong** [comment]
LGTM

**Comment by yujuhong** [comment]
@k8s-bot test this issue: #21484

**Comment by yujuhong** [comment]
@k8s-bot test this issue: #22655

**Comment by yujuhong** [comment]
It's sad that this PR is still waiting to be merged...

**Comment by pmorie** [comment]
@yujuhong :cry: :sob:

**Comment by derekwaynecarr** [comment]
This PR is high-value and needs to merge to help debug flakes.  I am labeling 1.3 milestone at least.

**Comment by pmorie** [comment]
:game_die: :game_die:

**Comment by pmorie** [comment]
@k8s-bot test this issue: #IGNORE

there's no way to determined at all what went wrong from the integration test spew

**Comment by pmorie** [comment]
:four_leaf_clover: :syringe:

**Comment by derekwaynecarr** [comment]
:beers: for the merge queue

**Comment by pmorie** [comment]
:dash: :game_die: :game_die:

**Comment by yujuhong** [comment]
Marking p1 to help deflake tests.

**Pull request merged**